databricks/koalas/tests/test_indexes.py

```diff
@@ -1001,3 +1001,64 @@ def test_unique(self):
 
         self.assert_eq(kmidx.unique().sort_values(), pmidx.unique().sort_values())
         self.assert_eq(kmidx.unique().sort_values(), pmidx.unique().sort_values())
+
+    def test_union(self):
+        # Index
+        pidx1 = pd.Index([1, 2, 3, 4])
+        pidx2 = pd.Index([3, 4, 5, 6])
+        kidx1 = ks.from_pandas(pidx1)
+        kidx2 = ks.from_pandas(pidx2)
+
+        self.assert_eq(kidx1.union(kidx2).sort_values(), pidx1.union(pidx2).sort_values())
+        self.assert_eq(kidx2.union(kidx1).sort_values(), pidx2.union(pidx1).sort_values())
+        self.assert_eq(
+            kidx1.union([3, 4, 5, 6]).sort_values(), pidx1.union([3, 4, 5, 6]).sort_values(),
+        )
+        self.assert_eq(
+            kidx2.union([1, 2, 3, 4]).sort_values(), pidx2.union([1, 2, 3, 4]).sort_values(),
+        )
+        # Sorting Index
+        if LooseVersion(pd.__version__) >= LooseVersion("0.24"):
+            # The `sort` argument is added in pandas 0.24.
+            self.assert_eq(kidx1.union(kidx2, sort=None), pidx1.union(pidx2, sort=None))
+            self.assert_eq(kidx2.union(kidx1, sort=None), pidx2.union(pidx1, sort=None))
+            self.assert_eq(
+                kidx1.union([3, 4, 5, 6], sort=None), pidx1.union([3, 4, 5, 6], sort=None),
+            )
+            self.assert_eq(
+                kidx2.union([1, 2, 3, 4], sort=None), pidx2.union([1, 2, 3, 4], sort=None),
+            )
```

Comment:
Since the default value for `sort` is `None`, do we need these?
Don't we need to check with other values?

Reply:
This is related with https://github.com/databricks/koalas/pull/1348#discussion_r394047937 .

I think we better use `False` as default, what do you think ??